File: backend/app/schemas/ocr.py
from pydantic import BaseModel, Field


class OcrBlock(BaseModel):
    text: str = Field(min_length=1)
    confidence: float = Field(ge=0, le=1)


class OcrResponse(BaseModel):
    ocr_text: str
    blocks: list[OcrBlock]
    corrected_text: str | None = None
    vision_text: str | None = None
    pix2text_text: str | None = None
    engine: str | None = None


class PageQuestionCandidate(BaseModel):
    question_no: str = ""
    question_text: str
    options: list[str] = []
    answer: str = ""
    solution: str = ""
    confidence: str = "medium"


class PageOcrResponse(BaseModel):
    page_text: str
    questions: list[PageQuestionCandidate]
    engine: str | None = None
    raw_text: str | None = None

File: frontend/src/api/ocr.ts
import { http } from './http'

/** OCR 含图片处理与可选 LLM 纠错，易超过默认 12s。 */
const OCR_HTTP_TIMEOUT_MS = 120_000

export interface OcrBlock {
  text: string
  confidence: number
}

export interface OcrResponse {
  ocr_text: string
  corrected_text?: string | null
  vision_text?: string | null
  blocks: OcrBlock[]
  pix2text_text?: string | null
  engine?: string | null
}

export interface PageQuestionCandidate {
  question_no: string
  question_text: string
  options: string[]
  answer: string
  solution: string
  confidence: 'high' | 'medium' | 'low' | string
}

export interface PageOcrResponse {
  page_text: string
  questions: PageQuestionCandidate[]
  engine?: string | null
  raw_text?: string | null
}

export function uploadOcrImage(file: File) {
  const formData = new FormData()
  formData.append('file', file)

  return http.post<OcrResponse>('/ocr', formData, {
    timeout: OCR_HTTP_TIMEOUT_MS,
  })
}

export function uploadPageOcrImage(file: File) {
  const formData = new FormData()
  formData.append('file', file)

  return http.post<PageOcrResponse>('/ocr/page', formData, {
    timeout: OCR_HTTP_TIMEOUT_MS,
  })
}
